src/components/functional/PaymentRequestTable/PaymentRequestTable.tsx

```diff
@@ -30,21 +37,72 @@ const PaymentRequestTable = ({ token, apiUrl = 'https://api.nofrixion.com/api/v1
     pageSize,
   );
 
+  const { deletePaymentRequest } = useDeletePaymentRequest(apiUrl, token);
+
   const localPaymentRequests: LocalPaymentRequest[] = paymentRequests.map((paymentRequest) =>
     RemotePaymentRequestToLocalPaymentRequest(paymentRequest),
   );
 
+  const onDeletePaymentRequest = async (paymentRequest: LocalPaymentRequest) => {
+    setPaymentRequestIdToDelete(paymentRequest.id);
+    setIsDeleteDialogOpen(true);
+  };
+
+  const onCopyPaymentRequestLink = async (paymentRequest: LocalPaymentRequest) => {
+    let link = `${apiUrl}/nextgen/pay/${paymentRequest.id}`;
+    await navigator.clipboard.writeText(link);
+
+    makeToast('success', 'Payment request link copied to clipboard.');
+  };
+
+  const closeDeleteDialog = () => {
+    setIsDeleteDialogOpen(false);
+  };
+
+  const confirmDeletePaymentRequest = async () => {
+    if (paymentRequestIdToDelete && paymentRequestIdToDelete !== '') {
+      let deleteError = await deletePaymentRequest(paymentRequestIdToDelete);
+
+      if (deleteError) {
+        makeToast('error', deleteError.title);
+        return;
+      }
+
+      makeToast('success', 'Payment request successfully deleted.');
+
+      await fetchPaymentRequests();
```

Comment:
This feels a little brute force. Would it be possible to not do this and instead delete the PaymentRequestRow?

Reply:
Thing is, if we do that, the current page shown will only be able to show 9 items until the user changes pages or refreshes. Do we want that?